src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1278,16 +1278,22 @@
   "Revoke the subject's certificate. Note this does not destroy the certificate.
    The certificate will remain in the signed directory despite being revoked."
   [{:keys [signeddir cacert cacrl cakey]} :- CaSettings
-   subject :- schema/Str]
-  (let [serial  (-> (path-to-cert signeddir subject)
-                    (utils/pem->cert)
-                    (utils/get-serial))
-        new-crl (utils/revoke (utils/pem->crl cacrl)
-                              (utils/pem->private-key cakey)
-                              (.getPublicKey (utils/pem->ca-cert cacert cakey))
-                              serial)]
-    (utils/crl->pem! new-crl cacrl)
-    (log/debug (i18n/trs "Revoked {0} certificate with serial {1}" subject serial))))
+   subjects :- [schema/Str]]
+  (let [original-crl (utils/pem->crl cacrl)
+        ca-private-key (utils/pem->private-key cakey)
+        ca-public-key (.getPublicKey (utils/pem->ca-cert cacert cakey))
+        new-crl (reduce
+                 (fn [crl subject]
```

Comment:
this makes a lot of sense, and I'm sure is a huge perf boost, but if we're not trying to get this in before the release we might want to look at creating a new `revokeMultiple` or something similar that takes a list of serials and loops over them calling `crl.addCRLEntry` each time so we aren't creating a new CRL for each serial. see https://github.com/puppetlabs/jvm-ssl-utils/blob/master/src/java/com/puppetlabs/ssl_utils/SSLUtils.java#L360

Reply:
Ah okay.

Reply:
I've opened https://github.com/puppetlabs/jvm-ssl-utils/pull/97 for this. However, that library is two major versions past what this 5.x branch has, but hasn't been branched yet. So we should talk about what is right there.

Reply:
I decided to branch that lib.